lib/index.js

```diff
@@ -11,79 +11,78 @@ const Hoek = require('hoek');
-    next();
-};
-
-
-exports.register.attributes = {
-    pkg: require('../package.json')
+        server.auth.scheme('basic', internals.implementation);
+    }
 };
 
 
 internals.implementation = function (server, options) {
 
     Hoek.assert(options, 'Missing basic auth strategy options');
-    Hoek.assert(typeof options.validateFunc === 'function', 'options.validateFunc must be a valid function in basic scheme');
+    Hoek.assert(typeof options.validate === 'function', 'options.validate must be a valid function in basic scheme');
 
     const settings = Hoek.clone(options);
 
     const scheme = {
-        authenticate: function (request, reply) {
+        authenticate: async function (request, h) {
+
+            const authorization = request.headers.authorization;
 
-            const req = request.raw.req;
-            const authorization = req.headers.authorization;
             if (!authorization) {
-                return reply(Boom.unauthorized(null, 'Basic', settings.unauthorizedAttributes));
+                return h.unauthenticated(Boom.unauthorized(null, 'Basic', settings.unauthorizedAttributes));
             }
 
             const parts = authorization.split(/\s+/);
 
             if (parts[0].toLowerCase() !== 'basic') {
-                return reply(Boom.unauthorized(null, 'Basic', settings.unauthorizedAttributes));
+                return h.unauthenticated(Boom.unauthorized(null, 'Basic', settings.unauthorizedAttributes));
             }
 
             if (parts.length !== 2) {
-                return reply(Boom.badRequest('Bad HTTP authentication header format', 'Basic'));
+                return h.unauthenticated(Boom.badRequest('Bad HTTP authentication header format', 'Basic'));
             }
 
-            const credentialsPart = new Buffer(parts[1], 'base64').toString();
+            const credentialsPart = Buffer.from(parts[1], 'base64').toString();
             const sep = credentialsPart.indexOf(':');
             if (sep === -1) {
-                return reply(Boom.badRequest('Bad header internal syntax', 'Basic'));
+                return h.unauthenticated(Boom.badRequest('Bad header internal syntax', 'Basic'));
             }
 
             const username = credentialsPart.slice(0, sep);
             const password = credentialsPart.slice(sep + 1);
 
-            if (!username && !settings.allowEmptyUsername) {
-                return reply(Boom.unauthorized('HTTP authentication header missing username', 'Basic', settings.unauthorizedAttributes));
-            }
+            if (!username &&
+                !settings.allowEmptyUsername) {
 
-            settings.validateFunc(request, username, password, (err, isValid, credentials) => {
+                return h.unauthenticated(Boom.unauthorized('HTTP authentication header missing username', 'Basic', settings.unauthorizedAttributes));
+            }
 
-                credentials = credentials || null;
+            try {
```

Comment:
This try scope is far too big. Declare the try variables with `var` and limit it to the one line.

Reply:
Correct me if I'm wrong but I don't think we need the try/catch at all then, if it's ok to just have the error rethrown in this scope.